chatbot-ui/components/UserInputDialog.tsx

```diff
@@ -0,0 +1,102 @@
+import { useEffect, useRef, useState } from 'react';
+import styles from '@/styles/Home.module.css';
+import LoadingDots from '@/components/LoadingDots';
+import { UseChatbotState } from '@/hooks/useChatbot';
+import ErrorSummary from '@/components/ErrorSummary';
+import classNames from 'classnames';
+
+const UserInputDialog = ({ sendMessage, error: APIError, fetching }: Props) => {
+  const [query, setQuery] = useState<string>('');
+  const [userInputError, setUserInputError] = useState<string | null>(null);
+  const textAreaRef = useRef<HTMLTextAreaElement>(null);
+
+  useEffect(() => {
+    textAreaRef.current?.focus();
+  }, []);
+
+  async function handleSubmit() {
+    if (!query) {
+      setUserInputError('Enter a question');
+      return;
+    }
+
+    setUserInputError(null);
+
+    const question = query.trim();
+
+    await sendMessage(question);
+    setQuery('');
+  }
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    if (e.key === 'Enter') {
+      e.preventDefault();
+      handleSubmit();
+    }
+  };
+
+  return (
+    <form
+      id="user-input-form"
+      onSubmit={(e) => {
+        e.preventDefault();
+        handleSubmit();
+        setQuery('');
+      }}
+    >
+      <div
+        className={classNames('govuk-form-group', {
+          'govuk-form-group--error': userInputError,
+        })}
+      >
+        <h1 className="govuk-label-wrapper">
+          <label className="govuk-label govuk-label--l" htmlFor="userInput">
+            What is your question?
+          </label>
+        </h1>
+        <div id="question-hint" className="govuk-hint">
+          Ask whatever question you like...
+        </div>
+
+        {userInputError && <ErrorSummary error={userInputError} />}
+        <textarea
+          className="govuk-textarea"
+          disabled={fetching || APIError !== null}
+          onKeyDown={handleKeyDown}
+          ref={textAreaRef}
+          rows={3}
+          maxLength={5000}
+          id="userInput"
+          name="userInput"
+          value={query}
+          onBlur={() => {
+            if (query) {
+              setUserInputError(null);
+            }
+          }}
+          onChange={(e) => {
+            setQuery(e.target.value);
+          }}
+          aria-describedby="question-hint"
+        />
+
+        <button
+          className="govuk-button"
+          disabled={fetching}
+          data-module="govuk-button"
+        >
+          Send
+          {fetching && (
+            <div className={styles.loadingwheel}>
+              <LoadingDots color="#000" />
+            </div>
+          )}
```

Comment:
This was something in the code before the GDS style was applied which has been broken now (or at least I can't see it). I think we can remove the LoadingDots component and its style now.

Reply:
I've removed it now.